core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2043,19 +2043,23 @@ struct ViewOffset<
       : m_dim(rhs.m_dim.N0, rhs.m_dim.N1, rhs.m_dim.N2, rhs.m_dim.N3,
               rhs.m_dim.N4, rhs.m_dim.N5, rhs.m_dim.N6, rhs.m_dim.N7),
         m_stride(rhs.stride_0()) {
-    if (((dimension_type::rank == 2)
-             ? rhs.m_stride.S1
-             : ((dimension_type::rank == 3)
-                    ? rhs.m_stride.S2
-                    : ((dimension_type::rank == 4)
-                           ? rhs.m_stride.S3
-                           : ((dimension_type::rank == 5)
-                                  ? rhs.m_stride.S4
-                                  : ((dimension_type::rank == 6)
-                                         ? rhs.m_stride.S5
-                                         : ((dimension_type::rank == 7)
-                                                ? rhs.m_stride.S6
-                                                : rhs.m_stride.S7)))))) != 1) {
+    bool right_stride_is_1 = true;
+    if (dimension_type::rank == 2)
+      right_stride_is_1 = (rhs.m_stride.S1 == 1);
+    else if (dimension_type::rank == 3)
+      right_stride_is_1 = (rhs.m_stride.S2 == 1);
+    else if (dimension_type::rank == 4)
+      right_stride_is_1 = (rhs.m_stride.S3 == 1);
+    else if (dimension_type::rank == 5)
+      right_stride_is_1 = (rhs.m_stride.S4 == 1);
+    else if (dimension_type::rank == 6)
+      right_stride_is_1 = (rhs.m_stride.S5 == 1);
+    else if (dimension_type::rank == 7)
+      right_stride_is_1 = (rhs.m_stride.S6 == 1);
+    else if (dimension_type::rank == 8)
+      right_stride_is_1 = (rhs.m_stride.S7 == 1);
+
```

Comment:
There is a reason that we wrote this as a big ternary operator. Can we verify that this produces the same code instructions at the end?